api/envoy/config/route/v3/route_components.proto

```diff
@@ -2247,6 +2247,21 @@ message HeaderMatcher {
   // * The regex ``\d{3}`` does not match the value *1234*, so it will match when inverted.
   // * The range [-10,0) will match the value -1, so it will not match when inverted.
   bool invert_match = 8;
+
+  // If specified, for any header match rule, if the header match rule specified header
+  // does not exist, this header value will be treated as empty. Defaults to false.
+  //
+  // Examples:
+  //
+  // * The header match rule specified header "header1" to range match of [0,
+  //   10], and invert_match is set to true; The "header1" header is not present.
+  //   The empty header does not match the range, so it will match when
+  //   inverted.
```

Comment:
It would be useful to point out how this behaves in both the invert and non-inverted case when `treat_missing_header_as_empy` is missing. It's pretty subtle for the reader to infer.

Reply:
PTAL at the new doc to see if it is good to go. Thankyou!